# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmJS.h
# Commit: 00d2290648a7
# Full Hash: 00d2290648a75aeefbb4738298ddfcda2d0797b0
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Add non-nullable references. r=lth
#   
#   This commit adds decoding, validation, and JS-API type-checking for
#   non-nullable references.
#   
# ==============================================================================

diff -r cd847eba7f9c -r 00d2290648a7 js/src/wasm/WasmJS.h
--- a/js/src/wasm/WasmJS.h	Tue Aug 18 00:06:00 2020 +0000
+++ b/js/src/wasm/WasmJS.h	Tue Aug 18 00:40:20 2020 +0000
@@ -167,7 +167,7 @@
 
 bool IsSharedWasmMemoryObject(JSObject* obj);
 
-// Check a value against the given reference type kind.  If the targetTypeKind
+// Check a value against the given reference type.  If the targetType
 // is RefType::Extern then the test always passes, but the value may be boxed.
 // If the test passes then the value is stored either in fnval (for
 // RefType::Func) or in refval (for other types); this split is not strictly
@@ -175,8 +175,8 @@
 //
 // This can return false if the type check fails, or if a boxing into AnyRef
 // throws an OOM.
-MOZ_MUST_USE bool CheckRefType(JSContext* cx, RefType::Kind targetTypeKind,
-                               HandleValue v, MutableHandleFunction fnval,
+MOZ_MUST_USE bool CheckRefType(JSContext* cx, RefType targetType, HandleValue v,
+                               MutableHandleFunction fnval,
                                MutableHandleAnyRef refval);
 
 }  // namespace wasm